system/include/emscripten/val.h

```diff
@@ -629,10 +630,17 @@ class val {
   friend struct internal::BindingType<val>;
 };
 
+// Declare a custom type that can be used in conjuction with
+// emscripten::register_type to emit custom TypeScript defintions for val types.
+#define EMSCRIPTEN_DECLARE_TYPE(name)                                          \
```

Comment:
This seems like a fairly generic name but its usage is very specific.  How about `EMSCRIPTEN_DECLARE_VAL_SUBTYPE`?   Or just `EMSCRIPTEN_DECLARE_VAL_TYPE`?